web/public/javascripts/routers/datasets.js

```diff
@@ -546,65 +465,27 @@ App.DatasetRoute = Ember.Route.extend({
       }
     });
 
-    var ownershipUrl = 'api/v1/datasets/' + id + "/owners";
-    $.get(ownershipUrl, function(data) {
-          if (data && data.status == "ok")
-          {
-            var owners = [];
-            if (data.owners && (data.owners.length > 0))
-            {
-              owners = data.owners;
-            }
-            controller.set("owners", owners);
-            setTimeout(function(){
-              $( "#sortedownertable" ).sortable({
-                start: function(event, ui) {
-                  var start_pos = ui.item.index();
-                  ui.item.data('start_pos', start_pos);
-                },
-                update: function( event, ui ) {
-                  var startPos = ui.item.data('start_pos');
-                  var endPos = ui.item.index();
-                  var originalOwners = controller.get("owners");
-                  if (!originalOwners || originalOwners.length == 0)
-                  {
-                    return;
-                  }
-                  var owners = originalOwners.slice();
-                  var updatedOwner = owners[startPos];
-                  owners.splice(startPos, 1);
-                  owners.splice(endPos, 0, updatedOwner);
-                  originalOwners.setObjects(owners);
-                  setTimeout(function(){
-                    setOwnerNameAutocomplete(controller)
-                  }, 500);
-                }
-              });
-
-              var allUserEntitiesUrl = 'api/v1/party/entities';
-              $.get(allUserEntitiesUrl, function(data) {
-                if (data && data.status == "ok")
-                {
-                  if (data.userEntities && (data.userEntities.length > 0))
-                  {
-                    var userEntitiesMaps = {};
-                    for (i=0; i < data.userEntities.length; i++)
-                    {
-                      userEntitiesMaps[data.userEntities[i].label] = data.userEntities[i].displayName;
-                    }
-                    var userEntitiesSource = Object.keys(userEntitiesMaps);
-                    controller.set("userEntitiesMaps", userEntitiesMaps);
-                    controller.set("userEntitiesSource", userEntitiesSource);
-                    setOwnerNameAutocomplete(controller);
-                  }
-                }
-              });
-            }, 500);
+    Promise.resolve($.get(`api/v1/datasets/${id}/owners`))
+        .then(({status, owners = []}) => (status === 'ok' && controller.set('owners', owners), void 0))
```

Comment:
Haha, it's not ES6 but fair enough, I was being stingy and saving two lines
I'll update but it's equivalent to
```javascript
.then(({status, owners = []}) => {
  controller.set('owners', owners);
  return undefined; // not necessary with function braces (all funcs return by default) but 1:1 translation of above
})
...
`